mrmustard/lab_dev/circuit_components.py

```diff
@@ -110,6 +110,22 @@ def __init__(
                 if self._representation:
                     self._representation = self._representation.reorder(tuple(perm))
 
+    @property
+    def _index_representation(self):
+        r"""
+        A hidden property that keeps track of representations for each index. If it does not exist already,
+        we initialize to Bargmann (B) on all indices.
+        """
+        if not hasattr(self, "_index_representation_"):
+            self._index_representation_ = {i: ("B", None) for i in self.wires.indices}
```

Comment:
Btw a nice convenience:
```suggestion
            self._index_representation_ = dict.fromkeys(self.wires.indices, ("B", None))
```